census/vanilla-tf/trainer/task.py

```diff
@@ -274,82 +194,98 @@ def parse_tf_config():
   task_index = tf_config_json.get('task').get('index')
 
   cluster_spec = tf.train.ClusterSpec(cluster)
-  return cluster_spec, job_name, task_index
-
-
-def read_input_tensors(file_name,
-                       batch_size=300,
-                       skiprows=None):
-  """Read the input file as an input queue.
-     See https://www.tensorflow.org/programmers_guide/reading_data/
-  """
-
-  filename_queue = tf.train.string_input_producer([file_name])
+  server = tf.train.Server(cluster_spec,
+                           job_name=job_name,
+                           task_index=task_index)
 
-  reader = tf.TextLineReader(skip_header_lines=skiprows)
-  key, value = reader.read(filename_queue)
-  value_columns = tf.expand_dims(value, -1)
+  if job_name == 'ps':
+    server.join()
+    return
+  elif job_name in ['master', 'worker']:
+    return run(server.target, job_name == 'master')
 
-  columns = tf.decode_csv(
-      value_columns,record_defaults=DEFAULTS)
 
+def parse_label_column(label_string_tensor): 
+  """Parses a string tensor into the label tensor
```

Comment:
Sorry - But when you do one hot life is good. Why lookup stuff and add more lines?